### DISCUSSION TITLE
Consume LittleHorse topics

### DISCUSSION THREAD

**Discussion by sauljabin** [sauljabin/kaskade#33] [General]
[LittleHorse](https://github.com/littlehorse-enterprises/littlehorse) is a high-performance microservice orchestration engine that allows developers to build scalable, maintainable, and observable applications.

LH uses kafka, and it serializes its topics with protobuf.

Install `protoc`:
```bash
brew install protobuf
```

Generate descriptor:
```bash
protoc --include_imports --descriptor_set_out=littlehorse.desc \
       --proto_path=${SCHEMAS_PATH}/internal \
       --proto_path=${SCHEMAS_PATH}/littlehorse \
       $(find ${SCHEMAS_PATH} -name "*.proto")
```

Read from `core-cmd` topic:
```bash
kaskade consumer -b localhost:9092 --from-beginning -t my-cluster-core-cmd \
        -k string -v protobuf \
        --protobuf descriptor=littlehorse.desc \
        --protobuf value=littlehorse.Command
```

Read from `metadata` topic:
```bash
kaskade consumer -b localhost:9092 --from-beginning -t my-cluster-metadata-cmd \
        -k string -v protobuf \
        --protobuf descriptor=littlehorse.desc \
        --protobuf value=littlehorse.MetadataCommand
```

**Comment by coltmcnealy-lh** [comment]
This is awesome, thanks @sauljabin !

**Reply by sauljabin** [reply]
Thanks!, this is available since kaskade 2.2.0. @eduwercamacaro maybe you will need this too.
